keras/layers/recurrent.py

```diff
@@ -893,7 +895,8 @@ def call(self, inputs, states, training=None):
             output = self.activation(output)
 
         # Properly set learning phase on output tensor.
-        if 0 < self.dropout + self.recurrent_dropout:
+        if (0 < self.dropout + self.recurrent_dropout or
+                getattr(states[0], '_uses_learning_phase', False)):
```

Comment:
Why do you need to check the `_uses_learning_phase` attribute of the first state?

Reply:
You're right, no need to check this condition. I must have been tested on the wrong code. Sorry about that.